backend/src/main/java/org/schmeckis/backend/service/RezeptService.java

```diff
@@ -3,25 +3,80 @@
 
-    public void createRezept(RequestRezept rezept) {
+    private final IdService idService;
+    public Rezept createRezept(RequestRezept rezept) {
         Optional<Rezept> isRezept = rezeptRepo.findByRezeptName(rezept.rezeptName());
         if (isRezept.isPresent()) {
             throw new IllegalArgumentException("Rezept already exists");
         }
-        rezeptRepo.save(new Rezept(rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), rezept.kategorieList()));
+        return rezeptRepo.save(new Rezept(idService.generateId(), rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), rezept.kategorieList()));
+    }
+
+    public Rezept getRezept(String id) throws IllegalArgumentException {
+        return rezeptRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Rezept not found"));
+    }
+
+    public void deleteRezept(String id) {
+        Rezept rezept = rezeptRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Rezept not found"));
+        rezeptRepo.delete(rezept);
+        /*rezeptRepo.findById(id).ifPresent(rezept1 -> {
+            throw new IllegalArgumentException("Rezept not deleted");
+        });*/
+    }
+
+    public Rezept updateRezept(String id, RequestRezept rezept) {
+        Rezept isRezept = rezeptRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Rezept not found"));
+        Rezept rezeptToUpdate = new Rezept(id, rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), rezept.kategorieList());
+        return rezeptRepo.save(rezeptToUpdate);
+    }
+
+    public Rezept addKategorieToRezept(String id, Kategorie kategorie) {
+        Rezept rezept = rezeptRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Rezept not found"));
+        List<Kategorie> kategorieList = new ArrayList<>(rezept.kategorieList());
+        kategorieList.add(kategorie);
+        Rezept rezeptToBeSafed = new Rezept(id, rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), kategorieList);
+        return rezeptRepo.save(rezeptToBeSafed);
+    }
+
+    public Rezept deleteKategorieFromRezept(String id, Kategorie kategorie) {
+        Rezept rezept = rezeptRepo.findById(id).orElseThrow(() -> new IllegalArgumentException("Rezept not found"));
+        List<Kategorie> kategorieList = new ArrayList<>(rezept.kategorieList());
+        kategorieList.remove(kategorie);
+        Rezept rezeptToBeSafed = new Rezept(id, rezept.rezeptName(), rezept.rezeptImageUrl(), rezept.rezeptKurzbeschreibung(), rezept.rezeptBeschreibung(), kategorieList);
+        return rezeptRepo.save(rezeptToBeSafed);
+    }
+
+    public List<Rezept> getAllRezepte() {
+        return rezeptRepo.findAll();
+    }
 
+    public List<Rezept> getRezepteByKategorie(String kategorieName) {
```

Comment:
Dies könnte performancemäßig eine Herausforderung werden. Ich vermute, dass man als Nutzer oft bestimmte Kategorien durch blättert, um sich daraus ein Rezept rauszusuchen. Also beispielhaft: Ich möchte einen Salat machen, also gucke ich in der Salat Kategorie nach. Auf jeden Fall erfordert dein Code, dass erstmal alle Rezepte geladen werden und dann wird sich jedes Rezept angeguckt und durch diese Kategorieliste geschaut. Hier bietet MongoDB auch eine Möglichkeit. Suche mal nach DBRef. Ich wollte es mir heute oder Montag ansehen.